contrib/node/tests/python/pants_test/contrib/node/tasks/test_node_resolve_src_deps.py

```diff
@@ -0,0 +1,361 @@
+                                    'projects/**/*'],
+                           package_name='workspace',
+                           package_manager='yarn',
+                           dependencies=[app])
+    return workspace
+
+  def _resolve_target(self, target):
+    context = self.context(target_roots=[target], options={
+      'npm-resolver': {'install_optional': False,
+                       'force_option_override': False,
+                       'install_production': False,
+                       'force': False,
+                       'frozen_lockfile': True}
+    })
+    self.wrap_context(context, [NodePaths])
+    task = self.create_task(context)
+    task.execute()
+
+    node_paths = context.products.get_data(NodePaths)
+    return task, node_paths
+
+  def test_resolve_simple_dep_graph(self):
+    dep = self._create_dep(provide_bin=True)
+    app = self._create_app(dep)
+    _, node_paths = self._resolve_target(app)
+ 
+    dep_node_path = node_paths.node_path(dep)
+    app_node_path = node_paths.node_path(app)
+    self.assertIsNotNone(dep_node_path)
+    self.assertIsNotNone(app_node_path)
+ 
+    # Verify that 'app/node_modules' has a correct symlink to 'dep'
+    app_node_modules_path = os.path.join(app_node_path, 'node_modules')
+    link_dep_path = os.path.join(app_node_modules_path, 'dep')
+    self.assertTrue(os.path.exists(app_node_modules_path))
+    self.assertTrue(os.path.islink(link_dep_path))
+
+    expected = os.path.relpath(dep_node_path, app_node_modules_path)
+    self.assertEqual(os.readlink(link_dep_path), expected)
+ 
+    # Verify that 'app/node_modules/.bin' has a correct symlink to 'dep'
+    app_bin_dep_path = os.path.join(dep_node_path, 'cli.js')
+    dep_bin_path = os.path.join(app_node_path, 'node_modules', '.bin')
+    link_dep_bin_path = os.path.join(dep_bin_path, 'dep')
+
+    self.assertTrue(os.path.islink(link_dep_bin_path))
+    relative_path = os.readlink(link_dep_bin_path)
+    expected = os.path.relpath(app_bin_dep_path, dep_bin_path)
+    self.assertEqual(relative_path, expected)
+ 
+  def test_resolve_symlink_bin_dict(self):
+    dep = self._create_dep(provide_bin=True, use_bin_dict=True)
+    app = self._create_app(dep)
+    _, node_paths = self._resolve_target(app)
+ 
+    dep_node_path = node_paths.node_path(dep)
+    app_node_path = node_paths.node_path(app)
+    self.assertIsNotNone(dep_node_path)
+    self.assertIsNotNone(app_node_path)
+ 
+    # Verify that 'app/node_modules/.bin' has a correct symlink to 'dep_cli'
+    app_bin_dep_path = os.path.join(dep_node_path, 'cli.js')
+    dep_bin_path = os.path.join(app_node_path, 'node_modules', '.bin')
+    link_dep_bin_path = os.path.join(dep_bin_path, 'dep_cli')
+    self.assertTrue(os.path.islink(link_dep_bin_path))
+    relative_path = os.readlink(link_dep_bin_path)
+    expected = os.path.relpath(app_bin_dep_path, dep_bin_path)
+    self.assertEqual(relative_path, expected)
+ 
+  def test_resolve_transitive_deps(self):
```

Comment:
This also invokes the transitive dep dependent project. Could you note that in the test name? Maybe `test_resolve_and_run_transitive_deps`